pages/Design-Patterns.md

```diff
@@ -68,7 +68,30 @@ yourself a lot of trouble down the road by using factories.
 When designing web applications, it often makes sense conceptually and architecturally to allow access to one and only
 one instance of a particular class. The singleton pattern enables us to do this.
 
-**TODO: NEED NEW SINGLETON CODE EXAMPLE**
+{% highlight php %}
+<?php
+class Singleton
+{
+    protected static $instance = null;
+
+    public static function getInstance() : self
+    {
+        if (null === static::$instance) {
+            static::$instance = new static();
+        }
+
+        return static::$instance;
+    }
+
+    protected function __construct() : void
```

Comment:
This should also be final to prevent extending objects from overriding the constructor with public.
`protected final function __construct() {}`